src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/EntrypointSelectionStep.cs

```diff
@@ -13,13 +14,18 @@ public class EntrypointSelectionStep : UpgradeStep
     {
         private readonly IPackageRestorer _restorer;
         private readonly IUserInput _userInput;
+        private readonly bool _isNonInteractiveMode;
+        private readonly string _entryPoint;
 
         public EntrypointSelectionStep(
+            UpgradeOptions options,
```

Comment:
I'm not a fan of passing in `UpgradeOptions` to things - this should be viewed more as an object for the console parsing. I know we're currently using it in a few steps, so maybe it's ok. At the moment, there are options that wouldn't matter potentially if hosted elsewhere, though

Reply:
We do need to pass the entry-point arg along, so it made sense to pass UpgradeOptions along similar to BackupStep